File: backend/schemas/health.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class HealthMeasurementBase(BaseModel):
    type: str # Heart Rate, SpO2, Blood Pressure, Steps, Sleep
    value: float
    secondary_value: Optional[float] = None
    unit: str
    source: str

class HealthMeasurementCreate(HealthMeasurementBase):
    pass

class HealthMeasurement(HealthMeasurementBase):
    id: int
    user_id: int
    recorded_at: datetime

    class Config:
        from_attributes = True

class AIAlertBase(BaseModel):
    alert_type: str
    message: str
    severity: str

class AIAlertCreate(AIAlertBase):
    pass

class AIAlert(AIAlertBase):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: mobile/lib/models/health.dart
class HealthMeasurement {
  final int id;
  final String type;
  final double value;
  final double? secondaryValue;
  final String unit;
  final String recordedAt;

  HealthMeasurement({
    required this.id,
    required this.type,
    required this.value,
    this.secondaryValue,
    required this.unit,
    required this.recordedAt,
  });

  factory HealthMeasurement.fromJson(Map<String, dynamic> json) {
    return HealthMeasurement(
      id: json['id'],
      type: json['type'],
      value: (json['value'] as num).toDouble(),
      secondaryValue: json['secondary_value'] != null ? (json['secondary_value'] as num).toDouble() : null,
      unit: json['unit'],
      recordedAt: json['recorded_at'],
    );
  }
}

class AIAlert {
  final int id;
  final String alertType;
  final String message;
  final String severity;

  AIAlert({
    required this.id,
    required this.alertType,
    required this.message,
    required this.severity,
  });

  factory AIAlert.fromJson(Map<String, dynamic> json) {
    return AIAlert(
      id: json['id'],
      alertType: json['alert_type'],
      message: json['message'],
      severity: json['severity'],
    );
  }
}
